packages/contracts/test/router/v6/nftx/listings.test.ts

```diff
@@ -0,0 +1,295 @@
+    );
+
+    const sushiPools = listings.filter(_ => _.vault).map(_ => _.vault)
+    const executions: ExecutionInfo[] = [
+      // 1. Fill listings
+      {
+        module: nftxModule.address,
+        data: nftxModule.interface.encodeFunctionData("c", [
+          listings.map((listing) => listing.order!.params),
+          {
+            fillTo: carol.address,
+            refundTo: carol.address,
+            revertIfIncomplete,
+            amount: totalPrice,
+          },
+          [
+            ...feesOnTop.map((amount) => ({
+              recipient: emilio.address,
+              amount,
+            })),
+          ],
+        ]),
+        value: totalPrice.add(
+          // Anything on top should be refunded
+          feesOnTop.reduce((a, b) => bn(a).add(b), bn(0)).add(parseEther("0.1"))
+        ),
+      },
+    ];
+
+    // Checks
+
+    // If the `revertIfIncomplete` option is enabled and we have any
+    // orders that are not fillable, the whole transaction should be
+    // reverted
+    if (
+      partial &&
+      revertIfIncomplete &&
+      listings.some(({ isCancelled }) => isCancelled)
+    ) {
+      await expect(
+        router.connect(carol).execute(executions, {
+          value: executions
+            .map(({ value }) => value)
+            .reduce((a, b) => bn(a).add(b), bn(0)),
+        })
+      ).to.be.revertedWith(
+        "reverted with custom error 'UnsuccessfulExecution()'"
+      );
+
+      return;
+    }
+
+    // Fetch pre-state
+
+    const ethBalancesBefore = await getBalances(
+      Sdk.Common.Addresses.Eth[chainId]
+    );
+
+    // Execute
+
+    const tx = await router.connect(carol).execute(executions, {
+      value: executions
+        .map(({ value }) => value)
+        .reduce((a, b) => bn(a).add(b), bn(0)),
+    });
+
+    // Fetch post-state
+
+    const ethBalancesAfter = await getBalances(
+      Sdk.Common.Addresses.Eth[chainId]
+    );
+
+    const poolBalanceAfter = await Promise.all(
+      sushiPools.filter(_ => _).map((address: any) => {
+        // return contract.getBalance(address)
+        return ethers.provider.getBalance(address);
+      })
+    )
+
+    const aliceOrderList = listings.filter(
+      ({ seller, isCancelled }) =>
+        !isCancelled && seller.address === alice.address
+    );
+
+    const aliceOrderSum = aliceOrderList
+      .map(({ price }) => bn(price))
+      .reduce((a, b) => bn(a).add(b), bn(0));
+
+    const bobOrderList = listings.filter(
+      ({ seller, isCancelled }) =>
+        !isCancelled && seller.address === bob.address
+    );
+
+    const bobOrderSum = bobOrderList
+      .map(({ price }) => bn(price))
+      .reduce((a, b) => bn(a).add(b), bn(0));
+    
+    
+    // Checks
+    const emilioBalance = ethBalancesAfter.emilio.sub(ethBalancesBefore.emilio);
+  
+    // Alice got the payment
+    // expect(aliceBalance).to.eq(aliceOrderSum);
+
+    // Bob got the payment
+    // expect(bobBalance).to.eq(bobOrderSum);
```

Comment:
Likewise, should probably enforce this.

Reply:
actually, `alice` and `bob` are the liquidity providers. 
so in this case, we check `carlo` spend is match the total order amount

Reply:
In this case maybe we can check the new LP balances of Alice and Bob?

Reply:
change to sushi pair balance check